### PR TITLE
Added pagination for event pages

### PR DESCRIPTION
**Pull request by cezaris13**
Added pagination for main event page. Used Skip and Take for pagination.
image

### CODE REVIEW

**Comment by cezaris13** [comment]
Do you have any suggestions how to add change number of events in the page in the frontend?

**Comment by Mart1nux** [comment]
> Do you have any suggestions how to add change number of events in the page in the frontend?

I've seen that matblazor has a choice chipset that could have like three predetermined options or, if I understand correctly how this works, put buttons into a layout grid with the same idea.

File: Pages/Pagination.razor.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+                return;
+
+            CurrentPage = link.Page;
+            await SelectedPage.InvokeAsync(link.Page);
+        }
+
+        private void LoadPages()
+        {
+            links = new List<LinkModel>();
+            bool isPreviousPageLinkEnabled = CurrentPage != 1;
```
Comment (ArentasMeinorius) [inline Pages/Pagination.razor.cs:33]: maybe isFirstPage would be easier to read

File: Pages/Pagination.razor.cs [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+            int previousPage = CurrentPage - 1;
+            links.Add(new LinkModel(previousPage, isPreviousPageLinkEnabled, "Previous"));
+
+            for (int i = 1; i <= TotaPagesQuantity; i++)
+            {
+                if (i >= CurrentPage - Radius && i <= CurrentPage + Radius)
+                    links.Add(new LinkModel(i) { Active = CurrentPage == i });
+            }
+
+            bool isNextPageLinkEnabled = CurrentPage != TotaPagesQuantity;
```
Comment (ArentasMeinorius) [inline Pages/Pagination.razor.cs:43]: isLastPage respectively

**Review by ArentasMeinorius — APPROVED** [review]

**Comment by cezaris13** [comment]
Added custom MatSelect custom page size:
image

**Comment by cezaris13** [comment]
> > Do you have any suggestions how to add change number of events in the page in the frontend?
> 
> I've seen that matblazor has a choice chipset that could have like three predetermined options or, if I understand correctly how this works, put buttons into a layout grid with the same idea.

Found the MatSelect exists, which does exactly what I wanted. Thanks for inspiration

**Review by Mart1nux — APPROVED** [review]

**Pull request merged**